src/schema/cdk-schema.ts

```diff
@@ -87,17 +87,24 @@ export function renderFullSchema(
     process.env.FORCE_COLOR = '0';
   }
 
-  // Find all constructs for which the props interface
-  // (transitively) only consists of JSON primitives or interfaces
-  // that consist of JSON primitives
+  // Find all constructs and top-level classes
   const constructType = typeSystem.findClass('constructs.Construct');
-  const constructs = typeSystem.classes.filter((c) => c.extends(constructType));
-
-  const deconstructs = constructs
-    .map(unpackConstruct)
-    .filter(
-      (c) => c && !isCfnResource(c.constructClass)
-    ) as ConstructAndProps[];
+  const cfnResourceType = typeSystem.findClass('aws-cdk-lib.CfnResource');
+
+  const constructs = typeSystem.classes.filter(
+    (c) => c.extends(constructType) && !c.extends(cfnResourceType)
+  );
+  const cdkObjects = typeSystem.classes.filter(
+    (c) =>
+      !c.extends(constructType) &&
+      c.initializer &&
+      c.initializer.parameters.length <= 1
```

Comment:
Isn't there any case where we have more than one constructor parameter?

Reply:
There are indeed. But I don't know we can represent them at the top-level.

Take this one for example
```ts
new HttpIntegration(url: string, props?: HttpIntegrationProps)
```
But Resources have to look like this:
```yaml
Type: aws-cdk-lib.aws_apigateway.HttpIntegration
Properties:
  Url: value
  Props: value
```
But then we have the issue why we dropped support for keyword args in the first place. 🤔 

--- 
Saying that, the check I currently have is not strict enough. We can only accept a struct here, nothing else. However if the additional arguments are optional we can drop them.

Reply:
What if we allow `Properties` to be assigned either a list or an object. Then, at type resolution, we make sure that lists can only be used for constructors parameters of CDK non-construct classes, and objects can only be used for constructor props and CFN resources.